dwave/preprocessing/composites/spin_reversal_transform.py

```diff
@@ -137,63 +137,100 @@ def _reorder_variables(sampleset: dimod.SampleSet,
+
+            Using the returned lowest energy state (the ground state) returned, we can
+            define a special SRT that transforms all programmed couplers to be ferromagnetic 
+            (ground state to all 1).
 
             >>> from dimod import ExactSolver
+            >>> import numpy as np
             >>> from dwave.preprocessing.composites import SpinReversalTransformComposite
             >>> base_sampler = ExactSolver()
             >>> composed_sampler = SpinReversalTransformComposite(base_sampler)
             ...
-            >>> Q = {('a', 'a'): -1, ('b', 'b'): -1, ('a', 'b'): 2}
-            >>> response = composed_sampler.sample_qubo(Q,
-            ...               num_spin_reversal_transforms=100)
-            >>> len(response)
-            400
+            >>> num_var = 6
+            >>> num_spin_reversal_transforms = 10
+            >>> J = {(i, i+1): np.random.random() for i in range(num_var-1)}
+            >>> h = {i: 0 for i in range(num_var)}
+            >>> response = composed_sampler.sample_ising(h, J,
+            ...               num_spin_reversal_transforms=num_spin_reversal_transforms)
+            >>> len(response) == 2**num_var * num_spin_reversal_transforms
+            True
+            >>> SRT = np.array([response.first.sample[i]!=1 for i in range(num_var)])
+            >>> response = composed_sampler.sample_ising(h, J,
+            ...               srts=SRT[np.newaxis,:], num_reads=1)
+            >>> sum(response.record.num_occurrences) == 2**num_var
+            True
         """
         sampler = self._child
 
         # No SRTs, so just pass the problem through
-        if not num_spin_reversal_transforms or not bqm.num_variables:
+        if num_spin_reversal_transforms==0 or not bqm.num_variables:
             sampleset = sampler.sample(bqm, **kwargs)
             # yield twice because we're using the @nonblocking_sample_method
             yield sampleset  # this one signals done()-ness
             yield sampleset  # this is the one actually used by the user
             return
 
+
+        # Get the srts matrix
+        if srts is None:
+            # We maintain the Leap behavior that num_spin_reversal_transforms == 1
+            # corresponds to a single problem with randomly flipped variables.
+            if num_spin_reversal_transforms is None:
+                num_spin_reversal_transforms = 1
+            srts = self.rng.random((num_spin_reversal_transforms, bqm.num_variables)) > .5
+        else:
+            nsrt, num_bqm_var = srts.shape
+            if num_bqm_var != bqm.num_variables:
+                raise ValueError('srt shape is inconsistent with the bqm')
+            if num_spin_reversal_transforms is not None:
+                if num_spin_reversal_transforms != nsrt:
+                    raise ValueError('srt shape is inconsistent with num_spin_reversal_transforms')
+            else:
+                num_spin_reversal_transforms = nsrt
```

Comment:
Consider an edge case:
```composite.sample(bqm, srts=np.zeros((0, bqm.num_variables))```

You should resolve the `srts` argument **before** the `num_spin_reversal_transforms == 0` check. And then you can revert the `== 0` change, and just test for truthy value as before.